boards: add new test supported feature usbd

jfischer-no opened the pull request:
We have recently added a new platform where the USB controller is only
supported in the new USB device support. This forces us to introduce a
new supported feature tag to distinguish the dependency. Use "usbd",
which is also the prefix of the new device support.

There are many boards that already have new driver support, but we will
replace the "usb_device" feature when the old device support is
deprecated.

Fixes: #73004

jfischer-no committed:
boards: add new test supported feature usbd

We have recently added a new platform where the USB controller is only
supported in the new USB device support. This forces us to introduce a
new supported feature tag to distinguish the dependency. Use "usbd",
which is also the prefix of the new device support.

There are many boards that already have new driver support, but we will
replace the "usb_device" feature when the old device support is
deprecated.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

jfischer-no committed:
samples: usb: use new usbd test feature

Use new feature "usbd" to distinguish the dependency.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

aescolar commented on boards/native/native_sim/native_sim.yaml:
```diff
@@ -15,6 +15,7 @@ supported:
   - eeprom
   - netif:eth
   - usb_device
+  - usbd
```
also for native_posix?

jfischer-no replied:
Actually it is not correct, there is not specific driver for native_sim or native_posix. I will try to rework it.

PerMac commented on tests/subsys/usb/device_next/testcase.yaml:
```diff
@@ -1,5 +1,5 @@
+  depends_on: usbd
   platform_allow:
     - native_sim
     - qemu_cortex_m3
```
what about this qemu? it is among platform_allow with depends_on: usbd but I don't see you adding such entry to the board in this PR. If it is not supported then I think it has to be taken out from platform_allow here and in uac2 sample, preferably replaced with some other qemu

jfischer-no replied:
Just no value to do it and I am not interested in running it elsewhere yet.

tmon-nordic commented:
What is really preventing us from deprecating the old USB stack? While it would be ideal to have the stable replacement available, is it necessary? The old stack has unfixable design issues and the sooner it is marked as deprecated the better IMHO.

jfischer-no committed:
tests: udc: rework driver test configuration

Distinguish between test case dependencies where a board has a test
feature "usbd" and where no dependency is required other than the
devicetree overlay file, as here to build the UDC skeleton driver.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

jfischer-no committed:
tests: usb: remove feature usb_device dependency

The tests do not depend on the usb_device or usbd feature.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>
